pom.xml

```diff
@@ -925,13 +916,11 @@
     <lz4.version>1.8.0</lz4.version>
     <snappy.version>1.1.10.4</snappy.version>
     <zstd-jni.version>1.5.5-2</zstd-jni.version>
-    <!--
-        Note that the version of protobuf shipped in hbase-thirdparty must match the version used
-        in hbase-protocol-shaded and hbase-examples. The version of jackson-[annotations,core,
-        databind] must be kept in sync with the version of jackson-jaxrs-json-provider shipped in
-        hbase-thirdparty.
+    <hbase-thirdparty.version>4.1.10-SNAPSHOT</hbase-thirdparty.version>
+    <!--Version of protobuf that hbase uses internally (we shade our pb)
+         Must match what is out in hbase-thirdparty include.
```

Comment:
replace with: 
"Version of protobuf that hbase uses internally (we shade our pb).
Must be kept in sync with the version of protobuf shipped in hbase-thirdparty"

Reply:
Why do we need to keep `internal.protobuf.version` at all?

Reply:
Could not remove the hardcoding of internal.protobuf.version as with BOM seems its not possible to pass on properties. And we actually use the protobuf version property in our code in `protobuf-maven-plugin` plugin

Reply:
That's super annoying.

Okay but maybe there's a hack. Can we use the bundle and process goals from https://maven.apache.org/plugins/maven-remote-resources-plugin/index.html to package up this property and then import it from the hbase-protocol-shaded build via https://www.mojohaus.org/properties-maven-plugin/read-project-properties-mojo.html ?

Reply:
> internal.protobuf.version

I tried this approach. Unfortunately this also did not work as "Properties used in plugin configurations must be available during the POM parsing phase." 

We can read/inject a property only after parsing is done; So the `compile-protoc` would still build with whatever is set as initial value for `internal.protobuf.version` in pom.xml

Reply:
>Maven load project model (from pom.xml) for project and all submodules at startup time. During project reading all available properties are resolved from projects and settings properties.
>
>This plugin (as all other) is executed in the later phase - when project model is already build in memory.
>
>So properties read from external files by this plugin can not by used in project definitions in items like <goal>, <version> and so on.

https://www.mojohaus.org/properties-maven-plugin/#plugin-limitations